src/data_conn.c

```diff
@@ -469,11 +472,12 @@ handle_epoll_in_event(replica_t *r)
 		rcomm_cmd->resp_list[idx].data_ptr = r->ongoing_io_buf;
 		if (rcomm_cmd->opcode == ZVOL_OPCODE_WRITE)
 			r->replica_inflight_write_io_cnt -= 1;
-		rcomm_cmd->resp_list[idx].status |= RECEIVED_OK;
 
 		if (rcomm_cmd->state != CMD_EXECUTION_DONE) {
+			rcomm_cmd->resp_list[idx].status |= RECEIVED_OK;
```

Comment:
Good to have comment with explanation why we set status in this way ....explain that recomm cleaner thread might free this struct if we set it before check state.